ios/engine/KMEI/KeymanEngine/Classes/Keyboard/InputViewController.swift

```diff
@@ -347,18 +401,11 @@ open class InputViewController: UIInputViewController, KeymanWebDelegate {
       perform(#selector(self.enableInputClickSound), with: nil, afterDelay: 0.1)
     }
 
-    var hasDeletedSelection = false
+    // `true` if there was selected text to be deleted
+    let deletedSelection = self.deleteSelection()
```

Comment:
We only need to do this call to `deleteSelection()` if we (a) there is a selection, and (b) we have no new text to insert, right? We really don't want to jump through this hoop if we don't have to.

Reply:
That's why the guard exists at the start of `deleteSelection`.  I placed it within `deleteSelection` rather than outside it.